components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -469,7 +469,7 @@ public void updateAPI(API.APIBuilder apiBuilder) throws APIManagementException {
                         (apiBuilder.getVersion())) && (originalAPI.getProvider().equals(apiBuilder.getProvider())) &&
                         originalAPI.getLifeCycleStatus().equalsIgnoreCase(apiBuilder.getLifeCycleStatus())) {
 
-                    if (apiBuilder.getPermission() != null && !("[]").equals(apiBuilder.getPermission())) {
+                    if (apiBuilder.getPermission() != null && !StringUtils.isEmpty(apiBuilder.getPermission())) {
```

Comment:
string utils check both null case and empty case. We can remove the null check :+1: